format.swift

```diff
@@ -0,0 +1,503 @@
+#!/usr/bin/env swift
+
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+// MARK: - Logging
+
+// It’s a little unfortunate that we need to duplicate all these functions from
+// swift-syntax-dev-utils. But executing format.swift as a script is significantly
+// faster than running it from a package using `swift run` (~1s vs ~15s), which
+// makes a big difference when running the script multiple times.
+
+func logProcessCommand(executableURL: URL?, arguments: [String]?) {
+  var message = ""
+
+  if let executableURL = executableURL {
+    message += executableURL.path
+  }
+
+  if let arguments = arguments {
+    message += " \(arguments.joined(separator: " "))"
+  }
+
+  print(message)
+}
+
+// MARK: - Paths
+
+enum Paths {
```

Comment:
The extension below this is `fileprivate` and this isn't 

I don't think that is a big issue. But what do you think of also making this fileprivate or removing below. 
Just for the sake of consistency? 

Same for the usage of `public` a bit further down?

Reply:
Good catch. It just doesn’t matter in a single file script but I made all types `internal` now for consistency.